File: kotlin/d2/ArraySum.kt
import java.io.*
import java.math.*
import java.security.*
import java.text.*
import java.util.*
import java.util.concurrent.*
import java.util.function.*
import java.util.regex.*
import java.util.stream.*
import kotlin.collections.*
import kotlin.comparisons.*
import kotlin.io.*
import kotlin.jvm.*
import kotlin.jvm.functions.*
import kotlin.jvm.internal.*
import kotlin.ranges.*
import kotlin.sequences.*
import kotlin.text.*

/**
Given an array of integers, find the sum of its elements.

For example, if the array , , so return .

Function Description

Complete the simpleArraySum function in the editor below. It must return the sum of the array elements as an integer.

simpleArraySum has the following parameter(s):

ar: an array of integers
Input Format

The first line contains an integer, , denoting the size of the array.
The second line contains  space-separated integers representing the array's elements.
*/

fun simpleArraySum(ar: List<Int>): Int {
    var total = 0
    for(i in ar) {
        total += i
    }
    return total
}

fun main(args: Array<String>) {
    
    val ar = listOf(1, 2, 3, 4, 10, 11)

    val result = simpleArraySum(ar)

    println(result)
}

File: go/d2/ArraySum.go
package main

import (
    "fmt"
)

/*
 * Complete the 'simpleArraySum' function below.
 *
 * The function is expected to return an INTEGER.
 * The function accepts INTEGER_ARRAY ar as parameter.
 */

func simpleArraySum(ar []int32) int32 {
    var total int32 = 0
		for _, it := range ar {
			total += it
		}	
		return total
}

func main() {

		ar := make([]int32, 4)
		ar[0] = 5
		ar[1] = 7
		ar[2] = 3
		ar[3] = 9

    result := simpleArraySum(ar)

    fmt.Printf("%d\n", result)
}
